cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -824,52 +825,83 @@ public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration
+                                      boolean injectStrimziMetricsReporter) {
+        OrderedProperties props = new OrderedProperties();
+        String configKey = "metric.reporters";
+        // the JmxReporter is explicitly added when metric.reporters is not empty
+        List<String> reporterClasses = List.of(
+                CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER,
+                "org.apache.kafka.common.metrics.JmxReporter",
+                StrimziMetricsReporterConfig.KAFKA_CLASS
+        );
+        List<Boolean> injectFlags = List.of(
+                injectCcMetricsReporter,
+                injectKafkaJmxReporter,
+                injectStrimziMetricsReporter
+        );
+        boolean hasUserMetricReporters = false;
+
+        for (int i = 0; i < reporterClasses.size(); i++) {
+            if (userConfig != null && !userConfig.getConfiguration().isEmpty() &&
+                    userConfig.getConfigOption(configKey) != null && injectFlags.get(i)) {
+                hasUserMetricReporters = true;
+                props.addPair(configKey, userConfig.getConfigOption(configKey));
+                userConfig.removeConfigOption(configKey);
+            }
+            if (injectFlags.get(i)) {
+                AbstractConfiguration.createOrAddListProperty(props, configKey, reporterClasses.get(i));
+            }
         }
-        if (injectStrimziMetricsReporter) {
-            AbstractConfiguration.createOrAddListConfig(userConfig, "metric.reporters", StrimziMetricsReporterConfig.KAFKA_CLASS);
+
+        if (!props.asMap().isEmpty()) {
+            printSectionHeader("Metric reporter configuration");
+            writer.println(hasUserMetricReporters ?
+                    "# Metric reporters configured by the user and Strimzi" :
+                    "# Metric reporters configured by Strimzi");
+            writer.println(props.asPairs());
+            writer.println();
         }
     }
 
-    /**
-     * Adds the Yammer kafka.metrics.reporters to the user configuration if the Strimzi Metrics Reporter is enabled.
-     *
-     * @param userConfig The user configuration to which the Yammer metrics reporter will be added.
-     * @param injectStrimziMetricsReporter Flag indicating whether to inject the Strimzi Metrics Reporter.
-     */
-    private void maybeAddYammerMetricsReporters(KafkaConfiguration userConfig, boolean injectStrimziMetricsReporter) {
+    private void printYammerMetricsReporters(AbstractConfiguration userConfig,
+                                             boolean injectStrimziMetricsReporter) {
+        OrderedProperties props = new OrderedProperties();
+        String configKey = "kafka.metrics.reporters";
+        boolean hasUserMetricReporters = false;
+
+        if (userConfig != null && !userConfig.getConfiguration().isEmpty() &&
+                userConfig.getConfigOption(configKey) != null && injectStrimziMetricsReporter) {
```

Comment:
Maybe it's similar to the previous question but:
if there is a "kafka.metrics.reporters" provided by the user BUT `injectStrimziMetricsReporter` is false, we are not entering in the `if` so we are not adding the user config to the `props` collection. The `hasUserMetricReporters` remain false and when we go down the final code with don't print any configuration. The result we are not printing the user provided configuration, or?